src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java

```diff
@@ -17,40 +18,87 @@
+        this.bCeIDServiceSoap = bCeIDServiceSoap;
         this.accountDetailsMapper = accountDetailsMapper;
     }
 
     @Override
-    public AccountDetails getAccountDetails(String userGuid) throws NestedEjbException_Exception {
+    public AccountDetails getAccountDetails(String userGuid, String bceidAccountType) throws NestedEjbException_Exception {
 
         if (StringUtils.isEmpty(userGuid)) return null;
 
+        AccountDetails accountDetails = getCsoDetails(userGuid);
+        if (null == accountDetails) {
+            accountDetails = getBCeidDetails(userGuid, bceidAccountType);
+        }
+
+        return accountDetails;
+    }
+
+    private AccountDetails getCsoDetails(String userGuid) throws NestedEjbException_Exception  {
+
         AccountDetails accountDetails = null;
+
         List<ClientProfile> profiles = accountFacadeBean.findProfiles(userGuid);
-        //An account must only one profile associated to proceed
+        // An account must only one profile associated to proceed
         if (profiles.size() == 1) {
-            accountDetails = accountDetailsMapper.toCsoAccountDetails(profiles.get(0), hasFileRole(userGuid));
+            accountDetails = accountDetailsMapper.toAccountDetails(profiles.get(0), hasFileRole(userGuid));
         }
         else if (profiles.size() > 1) {
             throw new CSOHasMultipleAccountException(profiles.get(0).getClientId().toString());
         }
 
         return accountDetails;
+    }
 
+    private AccountDetails getBCeidDetails(String userGuid, String accountType) {
+
+        AccountDetails accountDetails = null;
+        BCeIDAccountTypeCode accountTypeCode = getBCeIDAccountType(accountType);
+
+        if (accountTypeCode != BCeIDAccountTypeCode.VOID) {
+            AccountDetailRequest request = new AccountDetailRequest();
+            request.setOnlineServiceId("62B2-5550-4376-4DA7");
+            request.setRequesterUserGuid(userGuid);
+            request.setRequesterAccountTypeCode(accountTypeCode);
+            request.setUserGuid(userGuid);
+            request.setAccountTypeCode(accountTypeCode);
+            AccountDetailResponse response = bCeIDServiceSoap.getAccountDetail(request);
+            ResponseCode responseCode = response.getCode();
+
+            if (response.getCode() == ResponseCode.SUCCESS) {
+                accountDetails = accountDetailsMapper.toAccountDetails(response.getAccount());
+            }
+        }
+
+        return accountDetails;
     }
 
     public boolean hasFileRole(String userGuid) {
 
         UserRoles userRoles = roleRegistryPortType.getRolesForIdentifier("Courts", "CSO", userGuid, "CAP");
         List<RegisteredRole> roles = userRoles.getRoles();
         return roles != null && roles.stream().anyMatch(r -> r.getCode().equals("FILE"));
+    }
 
+    private BCeIDAccountTypeCode getBCeIDAccountType(String bceidAccountType) {
+        if (bceidAccountType.equals("Business")) return BCeIDAccountTypeCode.BUSINESS;
```

Comment:
Would a switch work here to reduce complexity? I forget how to program, feel free to slap me if this isn't the case.

Reply:
I'm actually in the process of switching this to a static lookup table right now... :)